src/named.rs

```diff
@@ -323,6 +323,31 @@ impl alloy_rlp::Decodable for NamedChain {
 #[allow(clippy::match_like_matches_macro)]
 #[deny(unreachable_patterns, unused_variables)]
 impl NamedChain {
+    /// Returns the number of decimals for the chain's native currency.
+    pub const fn native_currency_decimals(self) -> Option<u8> {
```

Comment:
hmm, this might be useful, but I'm not sure, it's definitely more to maintain.

wdyt @DaniPopes